application/backend/tests/integration/project_factory.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from datetime import datetime
+from uuid import uuid4
+
+from sqlalchemy.orm import Session
+
+from app.db.schema import DatasetItemDB, DatasetItemLabelDB, LabelDB, ModelRevisionDB, PipelineDB, ProjectDB
+from app.models import DatasetItemSubset
+
+
+class ProjectTestDataFactory:
+    """
+    Factory for creating test projects with related entities in the database.
+
+    This factory provides an interface for building complex test data scenarios involving projects and their related
+    entities (pipelines, models, labels, datasets). It handles proper relationship setup and database persistence.
+
+    The factory uses a builder pattern where you chain method calls to configure the project structure,
+    then call `build()` to persist everything to the database.
+
+    Usage:
+        Basic project:
+        >>> factory = ProjectTestDataFactory(db_session)
+        >>> project = factory.with_project(ProjectDB(name="Test")).build()
+
+        Project with pipeline and models:
+        >>> project = (
+        ...     ProjectTestDataFactory(db_session)
+        ...     .with_project(ProjectDB(name="Test"))
+        ...     .with_pipeline(is_running=True)
+        ...     .with_models([ModelRevisionDB(...)])
+        ...     .build()
+        ... )
```

Comment:
I like this factory class, however I'm not sure if I like the fact that `with_xyz` methods take complex objects/fixtures as argument. For example, `ModelRevisionDB` has a `project_id` which may differ from the one in `ProjectDB`, leading to internal inconsistency in the final project. It's powerful, we just need to be careful I guess. A possible way to mitigate inconsistencies between fixtures is to create global fixtures like `fxt_project_id` with a static UUID, and build the other fixtures on top of these predefined ids.

Reply:
We've started to produce some duplication in our integration tests around project setup, so I thought it would be helpful to introduce a factory pattern that can be reused. The `build` call should return the project along with all related entities, since they inherit the `project_id` of the project that's being built by the client.

I agree we should be cautious about extending this pattern further - it's a convention without any automated enforcement.